multiqc/plots/bargraph.py

```diff
@@ -189,7 +189,9 @@ def plot(data, cats=None, pconfig=None):
             not config.plots_force_interactive and len(plotsamples[0]) > config.plots_flat_numseries
         ):
             try:
-                return matplotlib_bargraph(plotdata, plotsamples, pconfig)
+                plot = matplotlib_bargraph(plotdata, plotsamples, pconfig)
+                report.num_mpl_plots += 1
+                return plot
```

Comment:
I think that you can just increment the counter before making the plot to avoid having to assign to a variable and then return that variable..

```suggestion
                report.num_mpl_plots += 1
                return matplotlib_bargraph(plotdata, plotsamples, pconfig)
```